src/main/java/com/barterbay/app/config/SecurityConfiguration.java

```diff
@@ -1,18 +1,57 @@
 package com.barterbay.app.config;
 
+import com.barterbay.app.service.UserService;
+import lombok.AllArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
 
-//TODO BB-21
+@Configuration
 @EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+@AllArgsConstructor
 public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+  private final UserService userService;
 
   @Override
   protected void configure(HttpSecurity http) throws Exception {
     http
+      .csrf().disable()
       .authorizeRequests()
-      .antMatchers("/ping/**")
-      .permitAll();
+      .antMatchers("/ping/**", "/js/**", "/css/**", "/images/**", "/fonts/**", "/vendor/**").permitAll()
+      .anyRequest().authenticated()
+      .and()
+      .formLogin()
+      .loginPage("/auth/login").permitAll()
+      .and()
+      .logout()
+      .invalidateHttpSession(true)
+      .clearAuthentication(true)
+      .deleteCookies("JSESSIONID");
+  }
+
+  @Override
+  protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+    auth.authenticationProvider(daoAuthenticationProvider());
+  }
+
+  @Bean
+  protected DaoAuthenticationProvider daoAuthenticationProvider() {
+    DaoAuthenticationProvider daoAuthenticationProvider = new DaoAuthenticationProvider();
```

Comment:
use var instead of `DaoAuthenticationProvider `
`DaoAuthenticationProvider daoAuthenticationProvider = new DaoAuthenticationProvider();` ->
`final var daoAuthenticationProvider = new DaoAuthenticationProvider();`

Reply:
corrected